iot-hub/Samples/device/FileUploadSample/FileUploadSample.cs

```diff
@@ -21,20 +21,92 @@ public FileUploadSample(DeviceClient deviceClient)
 
         public async Task RunSampleAsync()
         {
-            using (var fileStreamSource = new FileStream(FilePath, FileMode.Open))
+            const string filePath = "TestPayload.txt";
+
+            using var fileStreamSource = new FileStream(filePath, FileMode.Open);
+            var fileName = Path.GetFileName(fileStreamSource.Name);
+
+            Console.WriteLine($"Uploading file {fileName}");
+
+            var fileUploadTime = Stopwatch.StartNew();
+            
+            var fileUploadSasUriRequest = new FileUploadSasUriRequest
+            {
+                BlobName = fileName
+            };
+
+            // Note: GetFileUploadSasUriAsync and CompleteFileUploadAsync will use HTTPS as protocol regardless of the DeviceClient protocol selection.
+            Console.WriteLine("Getting SAS URI from IoT Hub to use when uploading the file...");
+            var sasUri = await _deviceClient.GetFileUploadSasUriAsync(fileUploadSasUriRequest);

```

Comment:
The sample fails here @timtay-microsoft. Any idea?

> '{"Message":"{\"errorCode\":400022,\"trackingId\":\"315c19c7c6fb495a9786e7dd0ae935c4-G:2-TimeStamp:09/05/2020 00:56:11\",\"message\":\"BadRequest\",\"timestampUtc\":\"2020-09-05T00:56:11.0393084Z\"}","ExceptionMessage":""}'

> Microsoft.Azure.Devices.Client.Transport.HttpClientHelper.ExecuteAsync(System.Net.Http.HttpMethod, System.Uri, System.Func<System.Net.Http.HttpRequestMessage, System.Threading.CancellationToken, System.Threading.Tasks.Task>, System.Func<System.Net.Http.HttpResponseMessage, bool>, System.Func<System.Net.Http.HttpResponseMessage, System.Threading.CancellationToken, System.Threading.Tasks.Task>, System.Collections.Generic.IDictionary<System.Net.HttpStatusCode, System.Func<System.Net.Http.HttpResponseMessage, System.Threading.Tasks.Task<System.Exception>>>, System.Threading.CancellationToken)

Reply:
I'm looking into this.

Reply:
I didn't have my hub configured properly. I fixed that and added instructions in program.cs.

Next failure happens when completing the file upload.

Reply:
> System.ArgumentException: '{"Message":"{\"errorCode\":400020,\"trackingId\":\"defa6fd07c2a48deb0ce6398fa452065-G:6-TimeStamp:09/08/2020 19:18:16\",\"message\":\"BadRequest\",\"timestampUtc\":\"2020-09-08T19:18:16.9608672Z\"}","ExceptionMessage":""}'

```
This exception was originally thrown at this call stack:
    Microsoft.Azure.Devices.Client.Transport.HttpClientHelper.ExecuteAsync(System.Net.Http.HttpMethod, System.Uri, System.Func<System.Net.Http.HttpRequestMessage, System.Threading.CancellationToken, System.Threading.Tasks.Task>, System.Func<System.Net.Http.HttpResponseMessage, bool>, System.Func<System.Net.Http.HttpResponseMessage, System.Threading.CancellationToken, System.Threading.Tasks.Task>, System.Collections.Generic.IDictionary<System.Net.HttpStatusCode, System.Func<System.Net.Http.HttpResponseMessage, System.Threading.Tasks.Task<System.Exception>>>, System.Threading.CancellationToken)
    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
    Microsoft.Azure.Devices.Client.Transport.HttpTransportHandler.CompleteFileUpload(Microsoft.Azure.Devices.Client.Transport.FileUploadCompletionNotification, System.Threading.CancellationToken)
    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
    Microsoft.Azure.Devices.Client.InternalClient.CompleteFileUploadAsync(Microsoft.Azure.Devices.Client.Transport.FileUploadCompletionNotification, System.Threading.CancellationToken)
    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
    Microsoft.Azure.Devices.Client.Samples.FileUploadSample.RunSampleAsync() in FileUploadSample.cs
    ...
    [Call Stack Truncated]
```

Reply:
Seems like an issue with the file that is being uploaded. When I run the sample with a full directory path to a different file (outside of the sample project), everything works as expected. Maybe just make the file path a parameter like the device connection string?

Reply:
I have no clue why only the last service call would throw an exception because of this, though